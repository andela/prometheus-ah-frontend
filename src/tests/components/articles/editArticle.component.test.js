import { shallow } from 'enzyme';
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedEditArticle,
{ EditArticle } from '../../../components/article/EditArticle';

const props = {
  editUserArticle: jest.fn().mockResolvedValue(Promise.resolve()),
  article: {
    Tags: [],
    User: {
      bio: 'I am a test',
      email: 'test@test.com',
      firstname: 'Testing',
      id: 5,
      image: 'string',
      lastname: 'Component',
      role: 'user',
      status: 'active',
      username: 'test'
    },
    title: 'new title',
    body: 'new body',
    description: 'new description',
    slug: 'new-article',
    createdAt: '2018-10-17T13:30:43.350Z',
    id: 83,
    readingTime: '1 min read',
    updatedAt: '2018-10-17T13:30:43.350Z',
    userId: 5
  },
  match: {
    params: {
      slug: '/new-article'
    }
  },
  location: { pathname: '/new-article' },
  history: {},
  update: {
    title: 'new edited',
    body: 'new edited body',
    description: 'new edited description'
  }
};
const mockStore = configureMockStore([thunk]);
const initialState = {
  articleReducer: {
    articles: [{
      title: 'new title',
      body: 'new body',
      description: 'new description',
      slug: 'new-1'
    }, {
      title: 'new title',
      body: 'new body',
      description: 'new description',
      slug: 'new-2'
    }],
  }
};
const store = mockStore(initialState);

describe('Edit article component', () => {
  it('should redirect to read article page correctly', () => {
    const wrapper = shallow(<EditArticle {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
  it('should display the necessary element', () => {
    const newProps = {
      editUserArticle: jest.fn(),
      article: {
        Tags: [],
        User: {
          bio: 'I am a test',
          email: 'test@test.com',
          firstname: 'Testing',
          id: 5,
          image: 'string',
          lastname: 'Component',
          role: 'user',
          status: 'active',
          username: 'test'
        },
        title: 'new title',
        body: 'new body',
        description: 'new description',
        slug: 'new-article',
        createdAt: '2018-10-17T13:30:43.350Z',
        id: 83,
        readingTime: '1 min read',
        updatedAt: '2018-10-17T13:30:43.350Z',
        userId: 5
      },
      match: {
        params: {
          slug: '/new-article'
        }
      },
      location: { pathname: '/new-article' },
      history: {},
    };
    const wrapper = shallow(
      <EditArticle {...newProps} />
    );
    expect(wrapper.find('div').exists()).toBe(true);
  });
  it('should call editArticle on submit', () => {
    const wrapper = shallow(<EditArticle {...props} />);
    const instance = wrapper.instance();
    instance.onSubmit();
    expect(props.editUserArticle).toHaveBeenCalled();
  });
  it('should set update to state after edit article succeed', () => {
    const wrapper = shallow(<ConnectedEditArticle store={store} {...props} />);
    expect(props.update).toEqual({
      body: 'new edited body',
      description: 'new edited description',
      title: 'new edited'
    });
    expect(wrapper.state()).toEqual({});
  });
});
