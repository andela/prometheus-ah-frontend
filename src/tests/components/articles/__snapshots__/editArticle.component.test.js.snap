// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit article component should redirect to read article page correctly 1`] = `
<div
  className="container"
>
  <div
    className="editor-config"
  >
    <Connect(ArticleForm)
      article={
        Object {
          "Tags": Array [],
          "User": Object {
            "bio": "I am a test",
            "email": "test@test.com",
            "firstname": "Testing",
            "id": 5,
            "image": "string",
            "lastname": "Component",
            "role": "user",
            "status": "active",
            "username": "test",
          },
          "body": "new body",
          "createdAt": "2018-10-17T13:30:43.350Z",
          "description": "new description",
          "id": 83,
          "readingTime": "1 min read",
          "slug": "new-article",
          "title": "new title",
          "updatedAt": "2018-10-17T13:30:43.350Z",
          "userId": 5,
        }
      }
      match={
        Object {
          "params": Object {
            "slug": "/new-article",
          },
        }
      }
      onSubmit={[Function]}
    />
  </div>
</div>
`;
